pandas/core/arrays/datetimes.py

```diff
@@ -374,6 +374,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif dtype == 'i8':
+            return self.asi8
```

Comment:
is this actually tested?, maybe use ``np.dtype('int64')`` here

Reply:
it is now, and done.